### PR TITLE
Dda basic

### PR DESCRIPTION
**Pull request by m00nbyt3**

### CODE REVIEW

File: include/dda.h [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+# include <stdio.h>
+# include <math.h>
+
+# include "libft.h"
+# include "angle.h"
+# include "point.h"
+
+typedef struct s_ray
+{
+	t_point	x_pos;
```
Comment (arikhativa) [inline include/dda.h:26]: i think x_pox and y_pox are not clear names
are you referring to the next y/x point?
if se then maybe call them next_x and next_y

File: include/angle.h [hunk 2]
```diff
@@ -6,20 +6,27 @@
...
 
 typedef struct s_angle
 {
 	double	radians;
+	double	min_in_range;
+	double	max_in_range;
 }				t_angle;
 
-#endif
+void	starting_angle(t_angle *angle, t_player *player);
```
Comment (arikhativa) [inline include/angle.h:28]: it is better to have a common function prefix for each class
starting_angle -> angle_starting
set_angle_range -> angle_set_range

File: include/dda.h [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+	t_point	y_pos;
+	double	slope;
+	double	intercept;
+	double	*incrementor;
+	t_angle	to_cast;
+}				t_ray;
+
+void	dda(char **map, t_player *p, t_ray	*ray_data, t_point *collision);
```
Comment (arikhativa) [inline include/dda.h:34]: dda -> dda_get_next_point
set_incrementor -> dda_set_incrementor

File: src/angle/angle.c [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+/*   Updated: 2023/02/22 18:03:41 by ycarro           ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "angle.h"
+
+void	starting_angle(t_angle *angle, t_player *player)
+{
+	if (player->dir == NORTH)
+		angle->radians = (3.0 / 2.0) * M_PI;
```
Comment (arikhativa) [inline src/angle/angle.c:18]: it is better to have macros in the code and remove numbers
can you name these value? 
3.0 / 2.0
1.0 / 2.0

File: include/dda.h [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+# include <stdlib.h>
+# include <stdio.h>
+# include <math.h>
+
+# include "libft.h"
+# include "angle.h"
+# include "point.h"
+
+typedef struct s_ray
```
Comment (arikhativa) [inline include/dda.h:24]: i think t_ray should be defined in a ray.h file so it is easy to find

File: src/dda/dda.c [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+{
+	(void)map;
+	(void)collision;
+	set_incrementor(ray_data, ray_data->to_cast.radians);
+	get_ray_data(ray_data, ray_data->to_cast.radians, p);
+}
+
+void	set_incrementor(t_ray *ray_data, double alpha)
+{
+	ray_data->incrementor = ft_calloc(2, sizeof(double));
```
Comment (arikhativa) [inline src/dda/dda.c:25]: i like to put all mem allocations in a ray_create() function and free them later in a ray_destroy() function
this way it is very easy to keep up with leaks
you can have a look at t_player for refrence

File: src/dda/dda.c [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+		ray_data->incrementor[1] = 1;
+	}
+	else
+	{
+		ray_data->incrementor[0] = 1;
+		ray_data->incrementor[1] = 1;
+	}
+}
+
+void	get_ray_data(t_ray *ray_data, double alpha, t_player *player)
```
Comment (arikhativa) [inline src/dda/dda.c:50]: why "get" is looks like "set"

**Review by arikhativa — CHANGES_REQUESTED** [review]

File: include/dda.h (hunk 1 above)
Comment (m00nbyt3) [reply include/dda.h]: They are called like that becouse they are like the actual position of the steps on x and y, but yes maybe I can find a better name

File: src/dda/dda.c (hunk 6 above)
Comment (m00nbyt3) [reply src/dda/dda.c:25]: I will try to change it, its true allocations should be controlled better

Comment (m00nbyt3) [reply src/dda/dda.c:25]: Need to talk about this in the next meeting

Comment (m00nbyt3) [reply src/dda/dda.c:25]: Changed

**Review by arikhativa — COMMENTED** [review]
what is this file:
minilibx_linux/test/mlx-test

**Pull request closed without merging**